centreon/src/Core/Host/Infrastructure/Repository/DbReadHostRepository.php

```diff
@@ -279,6 +281,39 @@ public function findParents(int $hostId): array
         return $statement->fetchAll(\PDO::FETCH_ASSOC);
     }
 
+    /**
+     * @inheritDoc
+     */
+    public function findNames(array $hostIds): HostNamesById
+    {
+        $concatenator = new SqlConcatenator();
+        $concatenator->defineSelect(
+            <<<'SQL'
+                SELECT DISTINCT(h.host_id), h.host_name
+                FROM `:db`.host h
+                WHERE h.host_id IN (:hostIds)
+                    AND h.host_register = '1'
+                SQL
+        );
+        $concatenator->storeBindValueMultiple(':hostIds', $hostIds, \PDO::PARAM_INT);
+
+        $statement = $this->db->prepare($this->translateDbName($concatenator->__toString()));
+        $concatenator->bindValuesToStatement($statement);
+        $statement->execute();
+
+        $groupNames = new HostNamesById();
+
+        while (false !== ($result = $statement->fetch(\PDO::FETCH_ASSOC))) {
```

Comment:
For maintainability (foreach => iterators) and performance (foreach is faster than while), you should use this syntax.
```suggestion
        $statement->setFetchMode(\PDO::FETCH_ASSOC);
        $statement->execute();

        $groupNames = new HostNamesById();

        foreach ($statement as $result) {
```